consensus/istanbul/backend/engine_test.go

```diff
@@ -644,6 +653,126 @@ func makeExpectedResult(indices []int, candidate []common.Address) []common.Addr
 	return copyAndSortAddrs(expected)
 }
 
+func TestSnapshot_MinimumStaking2(t *testing.T) {
+	type vote struct {
+		key   string
+		value interface{}
+	}
+	type expected struct {
+		blocks     []uint64
+		validators []int
+		demoted    []int
+	}
+	type testcase struct {
+		stakingInfo []uint64
+		votes       []vote
+		expected    []expected
+	}
+
+	testcases := []testcase{
```

Comment:
@jeongkyun-oh Could you please check below tests cases?

Reply:
@ehnuje The test cases are enhanced and updated. Please take another look.

Reply:
Thanks! It would be appreciated if the purpose of each test case is described.